modules/menu/menu.c

```diff
@@ -604,6 +604,8 @@ static void ua_event_handler(struct ua *ua, enum ua_event ev,
 		}
 
 		if (!str_cmp(call_id(call), menu.callid)) {
+			mem_deref(menu.attended_callid);
+			menu.attended_callid = NULL;
```

Comment:
You can write:
```
menu.attended_callid = mem_deref(menu.attended_callid);
```
And it is displaced here.

Reply:
Thanks for the reminder with `mem_deref`.

Can you elaborate why you think it is displaced?
If you are in the progress of transferring a call and the other side of your active call hung up, you are no longer in the progress of transferring a call.

Reply:
`if (!str_cmp(call_id(call), menu.callid))` means "if the call terminated is the current call".

I think you need another if statement:
```c
if (!str_cmp(call_id(call), menu.attended_callid)) {
     menu.attended_callid = mem_deref(menu.attended_callid);
}
```